File: today2/linked-list-cycle.ts
/**
 * https://leetcode.com/problems/linked-list-cycle/description/
 */
export { hasCycle, ListNode };

function hasCycle(head: ListNode | null): boolean {
  return false;
}

class ListNode {
  val: number;
  next: ListNode | null;
  constructor(val?: number, next?: ListNode | null) {
    this.val = val === undefined ? 0 : val;
    this.next = next === undefined ? null : next;
  }
}

File: returned/python/today11/linked_list_cycle.py
from typing import Optional

class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution:
    def hasCycle(self, head: Optional[ListNode]) -> bool:
        cache = set()

        while head:
            if head in cache:
                return True
            cache.add(head)
            head = head.next

        return False